csharp/device/Microsoft.Azure.Devices.Client/Message.cs

```diff
@@ -128,7 +158,11 @@ public string MessageId
         {
             get
             {
+#if NETMF
+                return (string)(this.GetSystemProperty(MessageSystemPropertyNames.MessageId) ?? string.Empty);
```

Comment:
isn't this going to throw an InvalidCastException instead of being null? I think the "as" keyword has the desired effect here more than a traditional cast (could be different with the Microframework, I'm not sure)